code/datums/elements/decals/_decal.dm

```diff
@@ -5,18 +5,18 @@
 	var/description
 	var/mutable_appearance/pic
 
-/datum/element/decal/Attach(atom/target, _icon, _icon_state, _dir, _cleanable=FALSE, _color, _layer=TURF_LAYER, _description, _alpha=255)
+/datum/element/decal/Attach(atom/target, _icon, _icon_state, _dir, _cleanable=FALSE, _color, _layer=TURF_LAYER, _plane=FLOOR_PLANE, _description, _alpha=255)
 	. = ..()
-	if(!isatom(target) || !generate_appearance(_icon, _icon_state, _dir, _layer, _color, _alpha, target))
+	if(!isatom(target) || !generate_appearance(_icon, _icon_state, _dir, _layer, _plane, _color, _alpha, target))
```

Comment:
You can absolutely have overlays with separate planes from the parent. If you couldn't the carpet wouldn't work and since it does I must conclude that you can.

Reply:
I am suspect of this, and am going to leave this unresolved for now.
If it works as is, you're probably right, but I'd prefer to get some confirmation first. I was under the impression that part of the point of vis_overlays was to allow things on different planes.